apps/mysql.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apps
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+)
+
+type LoggingProcessorMysqlError struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+}
+
+func (LoggingProcessorMysqlError) Type() string {
+	return "mysql_error"
+}
+
+func (p LoggingProcessorMysqlError) Components(tag string, uid string) []fluentbit.Component {
+	c := confgenerator.LoggingProcessorParseRegex{
+		// Format documented: https://dev.mysql.com/doc/refman/8.0/en/error-log-format.html
```

Comment:
It looks like the error log format for MySQL changed drastically in 8.x. MySQL 5.x and apparently MariaDB use a wildly different format:

https://mariadb.com/kb/en/error-log/#format

I'm not sure how to handle this but in general I think users expect MySQL and MariaDB to be interchangeable.  Maybe it's sufficient to call out in the docs "This receiver only supports Oracle MySQL >= 8.0 and does not support MySQL 5.x or MariaDB"?

Reply:
Does it make sense to have "supported version" as  flag or create different receivers for the different versions (with latest being the default) ?

I know we had internal discussion on this, @quentinmit @qingling128 what's the recent thinking here?

Reply:
I believe the recent thinking is "add a version field if needed, but try to avoid it". e.g. if we can write a regex that easily supports both, or two separate regexes that both apply (I think that's possible here?), or declare that we don't support the old version, that's preferable.

Oracle MySQL 8.0 is only 3 years old. RHEL 7 includes MariaDB (and presumably therefore the old log format). RHEL 8 includes both MariaDB and MySQL 8.0.

Reply:
From what I understand of Fluent Bit we can apply two Parsers to the same Filter and that should handle either log format. It'll require a bit of confgenerator updating but shouldn't be too bad. It will attempt to apply each regex to each line so the sooner they fail (more specific time format) the better for performance.

Reply:
Yes Fluent Bit support multiple parsers for different formats:

![image](https://user-images.githubusercontent.com/5287526/135684240-721e3561-4992-438f-a775-08f38067f0fa.png)

Reply:
Once https://github.com/GoogleCloudPlatform/ops-agent/pull/225 is approved/merged an additional parser can be added to support multiple error file formats.

Reply:
Yup! Working on that & query log parsers, currently on another branch in case this one was merged ahead, but otherwise intent to bring into this PR.